src/java/org/apache/cassandra/index/Index.java

```diff
@@ -426,6 +426,16 @@ public Indexer indexerFor(DecoratedKey key,
                               WriteContext ctx,
                               IndexTransaction.Type transactionType);
 
+    /**
+     * Tells whether this index supports replica fitering protection or not.
+     *
+     * @return true if this index supports replica filtering protection, false otherwise
+     */
+    default boolean supportsReplicaFilteringProtection()
```

Comment:
I wonder if it would make sense to use a more generic name for this method, something like `supportsFiltering`. @maedhroz wdyt?

Reply:
how about making it more generic? like `supports(IndexFeature feature)`. it will be useful for future indexes. WDYT?

Reply:
I would like Zhao's proposal if this was around an optional feature, but in some ways RFP is *required*, given an index implementation can't be correct without supporting it. (i.e. In a perfect world we wouldn't need the method at all.) In terms of naming, I'm fine with `supportsFiltering()` as long as the JavaDoc outlines RFP as an example of why we would need to filter.

Reply:
I dont agree with `supportsFiltering`, it is pretty non-telling ... like filtering of what? I would go with this name as is, without `IndexFeature` as argument. Deal?